app/storage/datastore.py

```diff
@@ -37,6 +36,9 @@ def __init__(self, client):
         self.client = client
 
     def put(self, model, overwrite=True):
+        if not overwrite:
+            raise Exception('Unique key checking not supported')
```

Comment:
Should a more specific error be raised?

Reply:
In what way?

Reply:
I'm not a fan of raising a bare `Exception` when a more specific subclass of it can be raised. Even if it is a custom exception class. In this instance `NotImplementedError` might be more applicable. Or don't give the option to pass `overwrite` at all.

Reply:
Not passing it would cause more issues as we have 3 storage providers matching same interface

Reply:
Then it sounds like NotImplementedError suits